src/cargo/sources/registry/mod.rs

```diff
@@ -322,10 +322,9 @@ impl<'cfg> Registry for RegistrySource<'cfg> {
         // theory the registry is known to contain this version. If, however, we
         // come back with no summaries, then our registry may need to be
         // updated, so we fall back to performing a lazy update.
-        if dep.source_id().precise().is_some() && !self.updated {
-            if try!(self.index.query(dep)).is_empty() {
+        if dep.source_id().precise().is_some() && !self.updated &&
+            try!(self.index.query(dep)).is_empty() {
                 try!(self.do_update());
```

Comment:
The indentation here is now wrong, can this stay as it was before? (more readable)